### PR TITLE
First draft of MDS high level design document.

### PR DESCRIPTION
**Pull request by stuwilkins**
This is the first draft of what we discussed on Friday...... I will call something for Tuesday to discuss. I will bring a blue tin hat........

### CODE REVIEW

File: source/metadatastore-format.rst [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+
+Events are the smallest quantum of data stored in the metadatastore. They group
+values which are associated with temporally identical data. The definition of
+"temporally identical" is determined by the DAQ system. For example, the 32
+channels in a scaler can be considered to be temporally identical because they
+are hardware synchronized. Inclusion of a CCD image (with a reference to the
+file store) can be included if this event is triggered at the same time, either
+by software or hardware.  Each ``event`` contains the data values and a client
+side timestamp of when the even was created. The same logic is applied to
+``trigger events``. These are grouped with temporally identical triggers (again
```
Comment (danielballan) [inline source/metadatastore-format.rst]: From meeting: make it "temporally bundled."

**Comment by danielballan** [comment]
Minor revisions from Jan. 20 meeting:

For any (*disaster-free!) run that's initiated, you should be guaranteed to get a `start_run` event and an `end_run` event.

The `beamline_config` will not be nested into the `start_event`. Many runs may point to the same `beamline_config`.

Major revision involving trigger events becoming event-wise start and end time.

File: source/metadatastore-format.rst [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+======
+
+Events are the smallest quantum of data stored in the metadatastore. They group
+values which are associated with temporally identical data. The definition of
+"temporally identical" is determined by the DAQ system. For example, the 32
+channels in a scaler can be considered to be temporally identical because they
+are hardware synchronized. Inclusion of a CCD image (with a reference to the
+file store) can be included if this event is triggered at the same time, either
+by software or hardware.  Each ``event`` contains the data values and a client
+side timestamp of when the even was created. The same logic is applied to
```
Comment (ericdill) [inline source/metadatastore-format.rst]: super major important change here...

typo in even (should be event)

File: source/metadatastore-format.rst [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+
+    end_run_event_c : {
+        "uid" : <id>,
+        "run_hdr" : <id>,
+        "reason" : <string>,
+        "time" : <time>
+    }
+
+The field ``reason`` can be used to describe why a run ended e.g. was it aborted or
+was there an exception during data collection. The field ``start_id`` is a
```
Comment (ericdill) [inline source/metadatastore-format.rst]: you wrote `start_id` but in the end_run_event such a field is not present

File: source/metadatastore-format.rst [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+        "keys" : {
+            "sclr" : {"source" : "PV:XF:23ID1-ES{Sclr:1}.CNT", "value" : 1 } 
+            "ccd" : {"source" : "PV:XF:23ID1-ES{Dif-Cam:PIMTE}cam1:Acquire",
+                     "value", 1 }
+        }
+        "run_hdr" : <uid>,
+        "time" : <time>
+    }
+
+Where as before ``run`` is a pointer to the start run event. The field
```
Comment (ericdill) [inline source/metadatastore-format.rst]: where is the `run` key?

File: source/metadatastore-format.rst [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+            "chan5" : {"value" : <value>, "timestamp" : <ts>},
+            "chan6" : {"value" : <value>, "timestamp" : <ts>},
+            "chan7" : {"value" : <value>, "timestamp" : <ts>},
+            "chan8" : {"value" : <value>, "timestamp" : <ts>},
+            "pimte" : {"value" : <uid>, "timestamp" : <ts>}
+        }
+        "time" : <time>
+    }
+
+Where the keys ``uid``, ``ev_desc`` and ``timestamp`` refer to the unique id, a
```
Comment (ericdill) [inline source/metadatastore-format.rst]: you wrote `timestamp` but the key is `time`

File: source/metadatastore-format.rst [hunk 6]
```diff
@@ -0,0 +1,221 @@
...
+.. _measure_events:
+
+Measure Events and Event Descriptors
+------------------------------------
+
+Measure events contain the data measured at a certain instance in time or
+explicit point in a sequence. Unlike :ref:`trigger_events` they contain actual
+data. For example::
+
+    measure_event_a : {
```
Comment (ericdill) [inline source/metadatastore-format.rst]: is the next event going to be called `measure_event_b`?

File: source/metadatastore-format.rst [hunk 7]
```diff
@@ -0,0 +1,221 @@
...
+Where the keys ``uid``, ``ev_desc`` and ``timestamp`` refer to the unique id, a
+link to the event descriptor and the EPICS timestamp respectively.
+
+The field ``seq_num`` can be used by step-wise data collection to determine the
+order of the events in a run, as in :ref:`trigger_events`.
+
+Event descriptors are used to describe an array of events which can form an
+event stream of a collection of events. For example a run forms
+event_descriptors at run start to define the data collected. For the example
+above ``event`` is described by the ``event_descriptor`` ::
```
Comment (ericdill) [inline source/metadatastore-format.rst]: maybe change this line to:

```
above ``measure_event_a`` is described by the ``event_desc_a`` ::
```

File: source/metadatastore-format.rst [hunk 8]
```diff
@@ -0,0 +1,221 @@
...
+Start and End Run Events
+------------------------
+
+The beginning of a data collection run creates an event which contains
+sufficient metadata and information to describe the data collection. For
+example, this is where beamline config information is located. The start run
+event also serves as a searchable entity which links all data associated by an
+event. For example::
+
+    start_event_b : {
```
Comment (ericdill) [inline source/metadatastore-format.rst]: Why are arbitrary letters being appended to these keys?

**Pull request merged**